Removal of alert system

kyuupichan opened the pull request:
Backport of bitcoin#7692 with minor BU-specific tweaks

laanwj committed:
Merge #7692: Remove p2p alert system

cfd519e Add release note documentation (BtcDrak)
6601ce5 protocol.h/cpp: Removes NetMsgType::ALERT (Thomas Kerin)
ad72104 Formatting (BtcDrak)
1b77471 Remove alert keys (BtcDrak)
01fdfef Remove `-alerts` option (BtcDrak)
9206634 Update alert notification and GUI (BtcDrak)
bbb9d1d Remove p2p alert handling (BtcDrak)

gandrewstone commented on src/main.cpp:
```diff
@@ -1827,6 +1825,26 @@ bool fLargeWorkForkFound = false;
 bool fLargeWorkInvalidChainFound = false;
 CBlockIndex *pindexBestForkTip = NULL, *pindexBestForkBase = NULL;
 
+static void AlertNotify(const std::string& strMessage, bool fThread)
```
what does alertNotify do? (add a comment)  I thought this PR removed alerts?

gandrewstone commented:
since this is 99% removal of code, I'm ok with the way this PR does not preserve the core author.  However, in general, please preserve attribution of the original author.  There are likely many ways to do this, but one is to cherry pick the core changes into your repo, then modify them to integrate with any BU changes.

Commit:
Remove BU-specific alert system bits

kyuupichan commented:
Reworked to keep upstream author; wasn't aware it had got dropped.  Probably because of the many conflicts and my attempts to resolve them.

Also added a comment as requested.  AlertNotify is for running external commands on various chain events and is still in 0.14.0.
